src/hooks/useServerSelect.js

```diff
@@ -0,0 +1,27 @@
+import apiClient from 'apis/apiClient';
+
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { doLogout } from 'reducers/auth.reducer';
+import { doInitializeCartList } from 'reducers/cart.reducer';
+import { AUTH_API_URL, ROUTES } from 'utils/constants';
+import { deleteCookie } from 'utils/cookie';
+
+const useServerSelect = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const setServerApi = (name = '이프') => {
+    localStorage.setItem('api_name', name);
+    apiClient.defaults.baseURL = AUTH_API_URL[name];
```

Comment:
직접 값에 할당하기보다는 apiClient에서 baseURL을 바꿀 수 있는 메서드를 가지고 있고 그걸 사용하도록 하는 게 어떨까요🙂?    
```suggestion
    apiClient.setBaseURL(AUTH_API_URL[name])
```    

apiClient의 정보를 요렇게 직접 접근해서 바꿔버리는 경우,    
나중에 어디에서 어떻게 변경하는 지를 추적하기 어려울 수 있을 것 같아요 🙂     
모듈로 묶었다면 해당 모듈의 정보는 모듈에서 제공하는 인터페이스를 통해서만 변경할 수 있게 해주는 게 어떨까 합니다~

Reply:
앗! 고민하던건데! 이렇게 할 수 있겠네요. 
apiClient를 class로 만들어서 setBaseURL과 setAuth 메서드를 추가했어요.
[refactor: loading 관련 네이밍을 동일하게 수정](https://github.com/woowacourse/react-shopping-cart-prod/pull/74/commits/58870934ffd2be950f967a57e89b2297d3efb1b7)